packages/components/src/snackbar/stories/index.tsx

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import type { ComponentMeta, ComponentStory } from '@storybook/react';
+
+/**
+ * Internal dependencies
+ */
+import Snackbar from '..';
+
+const meta: ComponentMeta< typeof Snackbar > = {
+	title: 'Components/Snackbar',
+	component: Snackbar,
+	argTypes: {
+		as: { control: { type: null } },
+		onRemove: {
+			action: 'onRemove',
+			control: { type: null },
+		},
+		onDismiss: {
+			action: 'onDismiss',
+			control: { type: null },
+		},
+		listRef: {
+			control: { type: null },
+		},
+	},
+	parameters: {
+		controls: {
+			expanded: true,
+		},
+		docs: { source: { state: 'open' } },
+	},
+};
+export default meta;
+
+const DefaultTemplate: ComponentStory< typeof Snackbar > = ( {
+	children,
+	...props
+} ) => {
+	return <Snackbar { ...props }>{ children }</Snackbar>;
+};
+
+export const Default: ComponentStory< typeof Snackbar > = DefaultTemplate.bind(
+	{}
+);
+Default.args = {
+	children:
+		'Use Snackbars to communicate low priority, non-interruptive messages to the user.',
+};
+
+export const WithActions: ComponentStory< typeof Snackbar > =
+	DefaultTemplate.bind( {} );
+WithActions.args = {
+	actions: [
+		{
+			label: 'Open WP.org',
+			url: 'https://wordpress.org',
+		},
+	],
+	children: 'Use Snackbars with an action link to an external page.',
+};
+
+export const WithIcon: ComponentStory< typeof Snackbar > = DefaultTemplate.bind(
+	{}
+);
+WithIcon.args = {
+	children: 'Add an icon to make your snackbar stand out',
+	icon: (
+		<span role="img" aria-label="Icon" style={ { fontSize: 21 } }>
+			🌮
+		</span>
+	),
```

Comment:
It would be cool if we could use `@wordpress/icons` like we do in other components (eg. [`Button`](https://github.com/WordPress/gutenberg/blob/0ad21d5efcbdcc074806c6252d3836c85a6c0ee9/packages/components/src/button/stories/index.js)), but I gave it a quick try and it looks like those icons are not well supported by the component currently (layout and color especially).

One more thing to improve on this component after it gets refactored to TypeScript

Reply:
Yeah, that'd be nice to simply have a `@wordpress/icon` as the icon.